### PR TITLE
Publicize scheduling: seperate published endpoint

### PR DESCRIPTION
**Pull request by artpi**
Use new endpoint to seperate scheduled from published.
This one: D4834-code

## Testing instructions

Go to post -> share, you should see this:
zrzut ekranu 2017-03-22 o 11 06 04

CC @lamosty @retrofox

### CODE REVIEW

File: client/my-sites/post-share/index.jsx [hunk 1]
```diff
@@ -249,33 +250,33 @@ class PostShare extends Component {
...
-					description={ translate( 'Live chat support and no advertising.' ) }
-					list={ [
-						translate( 'Live chat support' ),
-						translate( 'No Advertising' )
-					] }
-					plan={ PLAN_BUSINESS }
-					title={ translate( 'Upgrade to a Business Plan!' ) }
-				/>
-			);
+			return ( <Banner
```
Comment (lamosty) [inline client/my-sites/post-share/index.jsx]: Can we leave it like it was?

File: client/state/selectors/get-post-share-published-actions.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+import { mergeShareActionWithPublicizeConnections } from './get-post-share-scheduled-actions';
+
+/**
+ * Return a sharing-actions array propagaring data from publicize connections,
+ * creating a Date (moment) instance for the shareDate, etc.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @param {Number} postId Post ID
```
Comment (lamosty) [inline client/state/selectors/get-post-share-published-actions.js]: Maybe reword to say "Shared post ID"?

File: client/state/selectors/get-post-share-scheduled-actions.js [hunk 3]
```diff
@@ -50,3 +36,16 @@ export default function getSitePostPublicizeSharingActions( state, siteId, postI
 		};
 	} );
 }
+/**
+ * Return a sharing-actions array propagaring data from publicize connections,
+ * creating a Date (moment) instance for the shareDate, etc.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @param {Number} postId Post ID
```
Comment (lamosty) [inline client/state/selectors/get-post-share-scheduled-actions.js]: Maybe reword to say "Shared post ID"?

File: client/state/sharing/publicize/publicize-actions/actions.js [hunk 4]
```diff
@@ -14,22 +17,44 @@ import {
...
 	return ( dispatch ) => {
 		dispatch( {
-			type: PUBLICIZE_SHARE_ACTIONS_REQUEST,
+			type: PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST,
 			siteId,
 			postId,
 		} );
 
-		return new Promise( ( resolve, reject ) => wpcom.req.get( `/sites/${ siteId }/post/${ postId }/publicize/actions`, ( error, data ) => {
+		return new Promise( ( resolve, reject ) => wpcom.req.get( `/sites/${ siteId }/post/${ postId }/publicize/scheduled`, ( error, data ) => {
```
Comment (gwwar) [inline client/state/sharing/publicize/publicize-actions/actions.js]: wpcom already returns promises, so creating a promise here is unecessary

File: client/state/selectors/get-post-share-scheduled-actions.js [hunk 5]
```diff
@@ -9,23 +9,9 @@ import { moment } from 'i18n-calypso';
...
- */
-export default function getSitePostPublicizeSharingActions( state, siteId, postId ) {
-	const rawActions = get( state, [ 'sharing', 'publicize', 'sharePostActions', 'items', siteId, postId ], null );
-
-	if ( ! rawActions ) {
-		return null;
-	}
 
-	return map( rawActions, ( {
+export function mergeShareActionWithPublicizeConnections( state, postShareActions ) {
```
Comment (lamosty) [inline client/state/selectors/get-post-share-scheduled-actions.js]: Isn't there a file where we could put helper selectors like this? I mean, this helper selector is shared between `getPostShareScheduledActions` and `getPostSharePublishedActions` but this way, we put it into one of them which doesn't seem proper to me.

File: client/state/sharing/publicize/publicize-actions/actions.js [hunk 6]
```diff
@@ -14,22 +17,44 @@ import {
...
+			postId,
+		} );
+
+		return new Promise( ( resolve, reject ) => wpcom.req.get( `/sites/${ siteId }/post/${ postId }/publicize/published`, ( error, data ) => {
+			if ( error || ! data.items ) {
+				dispatch( { type: PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_FAILURE, siteId, postId, error } );
+				reject();
+			} else {
+				const actions = {};
+				data.items.forEach( action => ( actions[ action.ID ] = action ) );
```
Comment (gwwar) [inline client/state/sharing/publicize/publicize-actions/actions.js]: Map would be clearer here

File: client/state/sharing/publicize/publicize-actions/actions.js (hunk 4 above)
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/actions.js]: This line is too long, split please.

File: client/state/sharing/publicize/publicize-actions/actions.js [hunk 7]
```diff
@@ -14,22 +17,44 @@ import {
...
+
+export function fetchPostShareActionsPublished( siteId, postId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST,
+			siteId,
+			postId,
+		} );
+
+		return new Promise( ( resolve, reject ) => wpcom.req.get( `/sites/${ siteId }/post/${ postId }/publicize/published`, ( error, data ) => {
```
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/actions.js]: This line is too long, split please.

File: client/state/sharing/publicize/publicize-actions/reducer.js [hunk 8]
```diff
@@ -51,10 +54,20 @@ export const items = createReducer( {}, {
 
 }, publicizeActionsSchema );
 
-export const fetchingSharePostActions = createReducer( {}, {
-	[ PUBLICIZE_SHARE_ACTIONS_REQUEST_SUCCESS ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
-	[ PUBLICIZE_SHARE_ACTIONS_REQUEST_FAILURE ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
-	[ PUBLICIZE_SHARE_ACTIONS_REQUEST ]: ( state, { siteId, postId } ) => updateDataForPost( true, state, siteId, postId ),
+export const published = createReducer( {}, {
```
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/reducer.js]: This file contains multiple lines which are too long. Split them please.

File: client/state/sharing/publicize/publicize-actions/reducer.js [hunk 9]
```diff
@@ -51,10 +54,20 @@ export const items = createReducer( {}, {
...
+export const fetchingSharePostActionsScheduled = createReducer( {}, {
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_SUCCESS ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_FAILURE ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST]: ( state, { siteId, postId } ) => updateDataForPost( true, state, siteId, postId ),
+} );
+
+export const fetchingSharePostActionsPublished = createReducer( {}, {
+	[ PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_SUCCESS ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_FAILURE ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST]: ( state, { siteId, postId } ) => updateDataForPost( true, state, siteId, postId ),
```
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/reducer.js]: Missing space `[ PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST]`

File: client/state/sharing/publicize/publicize-actions/reducer.js [hunk 10]
```diff
@@ -51,10 +54,20 @@ export const items = createReducer( {}, {
...
-	[ PUBLICIZE_SHARE_ACTIONS_REQUEST_FAILURE ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
-	[ PUBLICIZE_SHARE_ACTIONS_REQUEST ]: ( state, { siteId, postId } ) => updateDataForPost( true, state, siteId, postId ),
+export const published = createReducer( {}, {
+	[ PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_SUCCESS ]: ( state, { siteId, postId, actions } ) => updateDataForPost( actions, state, siteId, postId ),
+}, publicizeActionsSchema );
+
+export const fetchingSharePostActionsScheduled = createReducer( {}, {
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_SUCCESS ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_FAILURE ]: ( state, { siteId, postId } ) => updateDataForPost( false, state, siteId, postId ),
+	[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST]: ( state, { siteId, postId } ) => updateDataForPost( true, state, siteId, postId ),
```
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/reducer.js]: Missing space `[ PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST]`

File: client/state/sharing/publicize/publicize-actions/reducer.js [hunk 11]
```diff
@@ -8,9 +8,12 @@ import { omit, get } from 'lodash';
...
 import {
-	PUBLICIZE_SHARE_ACTIONS_REQUEST,
-	PUBLICIZE_SHARE_ACTIONS_REQUEST_SUCCESS,
-	PUBLICIZE_SHARE_ACTIONS_REQUEST_FAILURE,
+	PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST,
+	PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_SUCCESS,
+	PUBLICIZE_SHARE_ACTIONS_SCHEDULED_REQUEST_FAILURE,
+	PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST,
+	PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_SUCCESS,
+	PUBLICIZE_SHARE_ACTIONS_PUBLISHED_REQUEST_FAILURE,
```
Comment (lamosty) [inline client/state/sharing/publicize/publicize-actions/reducer.js]: On line 27, there's a global `undefined`. According to our JS coding standards, we should not use globals and write `'undefined'` instead: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#globals

File: client/my-sites/post-share/index.jsx [hunk 12]
```diff
@@ -249,33 +250,33 @@ class PostShare extends Component {
...
+				title={ translate( 'Upgrade to a Business Plan!' ) }
+			/> );
 		}
-
-		return sharingActions.map(
-			( item, index ) => this.renderFooterSectionItem( item, index )
-		);
+		return ( <div>
+			<QuerySharePostActions siteId={ siteId } postId={ postId } status="scheduled" />
+			{ scheduledSharingActions.map( ( item, index ) => this.renderFooterSectionItem( item, index ) ) }
```
Comment (gwwar) [inline client/my-sites/post-share/index.jsx]: I'm getting infinite loaders from this line:
screen shot 2017-03-22 at 11 45 16 am

File: client/components/data/query-share-post-actions/index.jsx [hunk 13]
```diff
@@ -7,16 +7,20 @@ import { connect } from 'react-redux';
...
 
 class QuerySharePostActions extends Component {
 	static propTypes = {
 		siteId: PropTypes.number,
 		postId: PropTypes.number,
+		status: PropTypes.string,
 		// Connected props
-		isRequesting: PropTypes.bool.isRequired,
-		fetchPostShareActions: PropTypes.func.isRequired,
+		isRequestingScheduled: PropTypes.bool.isRequired,
```
Comment (lamosty) [inline client/components/data/query-share-post-actions/index.jsx]: I would not mark these as `isRequired` because they are passed from `connect` and don't have to be specified when using the component.

File: client/state/sharing/publicize/publicize-actions/actions.js (hunk 6 above)
Comment (artpi) [inline client/state/sharing/publicize/publicize-actions/actions.js]: Map will not work since it's not changing the keys and that's all what we do here - change the keys

File: client/components/data/query-share-post-actions/index.jsx [hunk 14]
```diff
@@ -7,16 +7,20 @@ import { connect } from 'react-redux';
...
+		isRequestingPublished: PropTypes.bool.isRequired,
+		fetchPostShareActionsScheduled: PropTypes.func.isRequired,
+		fetchPostShareActionsPublished: PropTypes.func.isRequired,
 	}
```
Comment (lamosty) [inline client/components/data/query-share-post-actions/index.jsx]: Semicolon missing ;)

File: client/my-sites/post-share/index.jsx (hunk 12 above)
Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: Why are we adding the query component here? Maybe it can be added into the `renderFooterSectionItem()` method.
About the bug, `scheduledSharingActions` is null so it couldn't be mapped.

Comment (artpi) [inline client/my-sites/post-share/index.jsx]: I think it was problem with old endpoint?

Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: it was with old selectors

Comment (lamosty) [inline client/my-sites/post-share/index.jsx]: > Why are we adding the query component here? Maybe it can be added into the renderFooterSectionItem() method.

It's not necessary as the query component loads all the scheduled and published shares at once so it can be loaded just once.
